tests/cluster_test.py

```diff
@@ -1,410 +1,1541 @@
+    while len(ports) < NODES_COUNT:
+        with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
+            try:
+                s.bind(('127.0.0.1', current_port))
+            except socket.error as e:
+                if e.errno == socket.errno.EADDRINUSE:
+                    current_port += 1
+                    continue
+                raise
+
+            ports.append(current_port)
+            current_port += 1
+
+    return ports
+
+
+@pytest.fixture(scope='module', autouse=True)
+def cluster_server(server_bin, free_ports):
+    server = setup_test_cluster(
+        free_ports, '/tmp/redis-cluster', server_exec=server_bin
+    )
+
+    yield server
+
+    server.terminate()
+    server.clear_directories()
+
+
+@pytest.fixture
+def nodes(free_ports):
+    return [
+        ('127.0.0.1', port)
+        for port in free_ports
+    ]
+
+
+@pytest.fixture
+def test_cluster(loop, nodes):
+    return loop.run_until_complete(
+        create_cluster(nodes, encoding='utf-8', loop=loop)
+    )
+
+
+@pytest.fixture
+def test_pool_cluster(loop, nodes):
+    pool_cluster = loop.run_until_complete(
+        create_pool_cluster(nodes, encoding='utf-8', loop=loop)
+    )
+
+    yield pool_cluster
+
+    loop.run_until_complete(pool_cluster.clear())
+
+
+@pytest.fixture
+def key_and_slot(test_cluster, loop):
+    loop.run_until_complete(test_cluster.set('key', 'value'))
+
+    yield 'key', KEY_KEY_SLOT
+
+    loop.run_until_complete(test_cluster.delete('key'))
+
+
+@pytest.fixture
+def zero_slot_key(test_cluster, loop):
+    loop.run_until_complete(test_cluster.set(SLOT_ZERO_KEY, 'value'))
+
+    yield SLOT_ZERO_KEY
+
+    loop.run_until_complete(test_cluster.delete(SLOT_ZERO_KEY))
+
+
+async def _wait_result(
+        func, cmp=lambda r: len(r) > 0, attempts=60, sleep_time=0.5, **kwargs
+):
+    attempts_count = 0
+    while attempts_count < attempts:
+        res = await func(**kwargs)
+        if cmp(res):
+            break
+
+        attempts_count += 1
+        await asyncio.sleep(sleep_time)
+    return res
+
+
+async def _clear_existing_slots(test_cluster, loop):
+    # We deon't know which node will be requested, so it's better to waite
+    # while changes will be applied to all cluster nodes
+    slots = await test_cluster.cluster_slots()
+    if not slots:
+        return
+
+    all_slots = [list(range(k[0], k[1] + 1)) for k in slots.keys()]
+    all_numbers = []
+    list(map(all_numbers.extend, all_slots))
+
+    await test_cluster.cluster_del_slots(*all_numbers, many=True, slaves=True)
+
+
+@pytest.fixture
+def reconfigure_cluster_server(cluster_server, test_cluster, loop):
+    yield
+    loop.run_until_complete(test_cluster.cluster_reset())
+    cluster_server.configure_cluster()
+    loop.run_until_complete(test_cluster.initialize())
+
+
+@pytest.fixture
+def clear_all_slots(test_cluster, loop, reconfigure_cluster_server):
+    loop.run_until_complete(_clear_existing_slots(test_cluster, loop))
+
+
+# Parse tests
```

Comment:
Instead of these comments I'd rather use classes to group tests (without any base classes).

Reply:
pytest style - it's to move them in a separate folder, in a separate files with conftest.
cluster /
    conftest.py
    parse_test.py
    ....